src/videos/src/videos-service/app.py

```diff
@@ -0,0 +1,339 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: MIT-0
+
+import logging
+import json
+import os
+import pathlib
+import pprint
+import subprocess
+import threading
+import time
+
+import boto3
+import srt
+from flask import Flask, jsonify, Response
+from flask_cors import CORS
+
+
+# -- Environment variables - defined by CloudFormation when deployed
+VIDEO_BUCKET = os.environ.get('RESOURCE_BUCKET')
+SSM_VIDEO_CHANNEL_MAP_PARAM = os.environ.get('PARAMETER_IVS_VIDEO_CHANNEL_MAP', 'retaildemostore-ivs-video-channel-map')
+USE_DEFAULT_IVS_STREAMS = os.environ.get('USE_DEFAULT_IVS_STREAMS') == 'true'
+
+DEFAULT_THUMB_FNAME = 'default_thumb.png'
+STATIC_FOLDER = '/app/static'
+STATIC_URL_PATH = '/static'
+SUBTITLE_FORMAT = 'srt'
+LOCAL_VIDEO_DIR = '/app/video-files/'
+
+# -- Parameterised ffmpeg commands
+FFMPEG_STREAM_CMD = """ffmpeg -loglevel panic -hide_banner -re -stream_loop -1 -i \"{video_filepath}\" \
+                           -r 30 -c:v copy -f flv rtmps://{ingest_endpoint}:443/app/{stream_key} -map 0:s -f {subtitle_format} -"""
+FFMPEG_SUBS_COMMAND = "ffmpeg -i \"{video_filepath}\" \"{subtitle_path}\""
+
+# -- Default stream details
+# These details will be returned by the API if the 'use default IVS streams' option is chosen for deployment.
```

Comment:
Where do the playback URLs come from? Who owns and maintains them? Can they change?

Reply:
We were asked to ensure that streams containing realistic content (ie. the vlog-style videos we have purchased) would be available to all users. Due to licensing issues, we are unable to distribute the video files to other users of the demo so we have been asked to deploy persistent streams which all users can consume using the IVS ingest URL. The AWS Retail team will maintain them going forward. 

The streams can change if there is a full redeployment of the underlying CloudFormation stack. In this case the default streams would need to be updated here. However, stream URLs would not change on new container deployments (we are using the same CD pipeline as is used in this repo).

The AWS retail team is in the process of obtaining a domain for the IVS demo; we could possibly use that to make the default URLs persistent. One for discussion on a call.

Reply:
Does it make sense to externalize the default stream details in a JSON file or something? Otherwise we're going to have to make changes to source when the product IDs change or when new/different default streams are added. Product IDs will be changing to UUIDs soon. I get that the defaults shouldn't change often but having the hardcoded makes it harder to maintain.

Reply:
We will make this change. Default streams will be picked up from the videos/ folder where user-provided videos may be provided.